slasher/rpc/server.go

```diff
@@ -32,6 +32,28 @@ type Server struct {
 	proposeLock     sync.Mutex
 }
 
+// HighestAttestations returns thee highest observed attestation source and epoch for a given validator id.
+func (ss *Server) HighestAttestations(ctx context.Context, req *slashpb.HighestAttestationRequest) (*slashpb.HighestAttestationResponse, error) {
+	ret := make([]*slashpb.HighestAttestation, 0)
+	for _, id := range req.ValidatorIds {
```

Comment:
add a context error check in the loop.

Reply:
with timeout?

Reply:
the context usually will have a deadline associated with it

Reply:
Yep the upstream context will typically have a deadline or some other cancelation method. All we have to do is check for if ctx.Err() != nil in this function's range loop